amy/workshops/migrations/0259_remove_deprecated_training_requirements.py

```diff
@@ -0,0 +1,87 @@
+# Generated by elichad on 2023-05-12 13:40
+
+from django.db import migrations
+from django.db.models import Q
+
+
+def rename_discussion_to_welcome_session(apps, schema_editor) -> None:
+    TrainingRequirement = apps.get_model("workshops", "TrainingRequirement")
+
+    try:
+        requirement = TrainingRequirement.objects.get(name="Discussion")
+        requirement.name = "Welcome Session"
+        requirement.save()
+    except TrainingRequirement.DoesNotExist:
+        pass
+
+
+def rename_welcome_session_to_discussion(apps, schema_editor) -> None:
+    TrainingRequirement = apps.get_model("workshops", "TrainingRequirement")
+
+    try:
+        requirement = TrainingRequirement.objects.get(name="Welcome Session")
+        requirement.name = "Discussion"
+        requirement.save()
+    except TrainingRequirement.DoesNotExist:
+        pass
+
+
+def delete_outdated_requirements(apps, schema_editor) -> None:
+    TrainingRequirement = apps.get_model("workshops", "TrainingRequirement")
+    TrainingProgress = apps.get_model("workshops", "TrainingProgress")
+
+    # migrate SWC/DC/LC specific progress to generic demo/lesson contribution
+    # this was already done in production AMY but not in local databases
```

Comment:
What do you mean by "local databases"? Do you mean development databases, or test-amy2 database?

This migration looks reasonable (except for lines 52-54 where Training Requirements are removed), but I'm trying to understand if it's truly needed.

Reply:
Development databases. When fake training progresses are created, they choose from all available TrainingRequirements, so my local dev database was populated with lots of SWC Demos, LC Homeworks, etc.
We don't have auto-generated fake data on test-amy2 so this migration isn't required there (though it would have been required if we had generated some).

Reply:
I've removed the creation/deletion of TrainingRequirements from this step of the migration since that's covered by the seeding, but this section still remains for discussion.